judge/jinja2/submission.py

```diff
@@ -18,7 +18,9 @@ def submission_layout(submission, profile_id, user, completed_problem_ids, edita
         can_view = True
     elif submission.problem_id in completed_problem_ids:
         can_view = submission.problem_id in tester_problem_ids
-        if not settings.DMOJ_PRIVATE_SUBMISSION:
+        if settings.DMOJ_SUBMISSION_SOURCE_VISIBILITY == 'all-solved':
             can_view = can_view or submission.problem.is_public
+        elif settings.DMOJ_SUBMISSION_SOURCE_VISIBILITY == 'only-own':
```

Comment:
I think we don't need this branch, since it's a no-op. On the other hand, it'd be good to have a new `settings.DMOJ_SOURCE_VISIBILITY == 'all'` branch at the top of this if/else tree `return True`.

Reply:
I agree.
I also worried about the line the most.
The purpose was to inform the existence of the branch in case somebody edits it in the future, but I know that it is two lines of unnecessary code.
Now that I wanted to match the branch order in the jinja2 and models source code, I think how about removing the 2 lines.
If you agree, I will only remove 2 unnecessary lines.

Reply:
I think it's better to remove these two lines. Your concern is totally valid but the proper solution is writing some unit tests.

Reply:
I understand.
Thank you for paying attention to the small suggestions and reviewing the code.